osu.Game/Rulesets/Scoring/ScoreProcessor.cs

```diff
@@ -253,7 +258,7 @@ private void updateScore()
         /// <returns>The total score in the given <see cref="ScoringMode"/>.</returns>
         public double ComputeFinalScore(ScoringMode mode, ScoreInfo scoreInfo)
         {
-            extractFromStatistics(scoreInfo.Ruleset.CreateInstance(),
+            extractFromStatistics(ruleset,
```

Comment:
Maybe worth adding sanity checks to ensure `ScoreInfo`'s ruleset matches the one that the score processor received in ctor?